controllers/discovery/initDiscoveryUserData.js

```diff
@@ -9,11 +9,30 @@ const {QueryTypes} = require('sequelize');
  * @returns
  */
 const InitDiscoveryUserData = async (req, res) => {
-  const {limit = 50, page = 0} = req.body;
+  const {limit = 50, page = 0} = req.query;
 
   const {userId} = req;
 
   try {
+    let totalDataQuery = `
+    SELECT 
+        count(A.user_id) as total_data
+    FROM users A
+    LEFT JOIN 
+        (SELECT 
+            common.*,
+            joint.common,
+        CASE WHEN joint.source = :userId THEN 1 ELSE 0 END as user_match
+        FROM users common
+        JOIN 
+            vwm_user_common_follower_count joint
+        ON common.user_id = joint.target
+        WHERE joint.source = :userId AND common.is_anonymous = false) CommonUsers
+    ON CommonUsers.user_id = A.user_id
+    LEFT JOIN user_follow_user B
+    ON A.user_id = B.user_id_followed AND B.user_id_follower = :userId
+    WHERE A.user_id != :userId AND A.is_anonymous = false AND A.is_banned = false`;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This SQL query is quite complex and might be difficult to maintain. Consider breaking it down into smaller parts or using an ORM to make it more readable and maintainable. Also, it's important to note that this query could potentially have performance issues depending on the size of your database and the indexes you have set up. Make sure to test it thoroughly with realistic data sizes.



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
@n-permana no need for this complex query, just count `users` table